packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -878,7 +879,12 @@ function useMemoCache(size: number): Array<any> {
 
   let data = memoCache.data[memoCache.index];
   if (data === undefined) {
-    data = memoCache.data[memoCache.index] = new Array(size);
+    data = memoCache.data[memoCache.index] = new Array(size).fill(
+      MEMO_CACHE_UNSET_SENTINEL,
+    );
+    // Attach sentinel to the array, eventually we can make the sentinel a public
+    // export from React and avoid modifying the array
+    (data: any)._ = MEMO_CACHE_UNSET_SENTINEL;
```

Comment:
Yea, I originally chose the `$._` approach merely to ease the implementation of codegen and patching. It's likely to be faster if we can directly export this symbol to not only save one hidden class transition/modification of array but also one property lookup in the runtime.

The only caveat I had thought about is that we need to deal with potential name clashes *everywhere* which is hairy:

```js
import { MEMO_CACHE_UNSET_SENTINEL as _ } from "react"

function Foo() {
  // ok
}

function Bar() {
  let _ = otherThing;
  // well
}
```